src/common/transformations/src/transformations/common_optimizations/shared_ops_optimization.cpp

```diff
@@ -107,6 +123,7 @@ bool shared_node_optimization(const shared_ptr<Model>& model) {
     const auto& order = model->get_ordered_ops();
     for (size_t i = 0; i < order.size(); ++i)
         index_map[order[i]] = i;
+    std::unordered_map<std::shared_ptr<ov::Node>, ov::AnyMap> node_attributes_cache;
```

Comment:
Consider use raw pointer as key instead copy shared pointer, it looks like local instance take of ownership maybe not required.

Reply:
I agree. This should be taken care of in the later PRs